### PR TITLE
Update Admin CLI shard operations to support SQL databases

### PR DESCRIPTION
**Pull request by Shaddoll**
**What changed?**
Update Admin CLI shard operations to support SQL databases

**Why?**
To address https://github.com/uber/cadence/issues/2777

**How did you test it?**
local test with mysql

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: tools/cli/adminTimers.go [hunk 1]
```diff
@@ -32,15 +32,13 @@ import (
...
-	"github.com/uber/cadence/common/log/loggerimpl"
 	"github.com/uber/cadence/common/persistence"
-	"github.com/uber/cadence/common/persistence/cassandra"
-	"github.com/uber/cadence/common/quotas"
 )
 
 // Loader loads timer task information
 type Loader interface {
 	Load() []*persistence.TimerTaskInfo
+	Close()
```
Comment (mantas-sidlauskas) [inline tools/cli/adminTimers.go:41]: Nit: Interface name and comment is now misleading. Ref https://golang.org/doc/effective_go#interface-names

File: tools/cli/adminDBScanCommand.go [hunk 2]
```diff
@@ -316,6 +316,25 @@ func initializeHistoryManager(c *cli.Context) persistence.HistoryManager {
...
+	case "cassandra":
+		shardStore = initializeCassandraShardStore(c, "current-cluster", logger)
+	default:
+		if sql.PluginRegistered(dbType) {
+			shardStore = initializeSQLShardStore(c, "current-cluster", logger)
+		} else {
+			supportedDBs := append(sql.GetRegisteredPluginNames(), "cassandra")
+			ErrorAndExit(fmt.Sprintf("The DB type is not supported. Options are: %s.", supportedDBs), nil)
+		}
+	}
```
Comment (mantas-sidlauskas) [inline tools/cli/adminDBScanCommand.go:333]: Nit: for a better readability, I would check if dbType is valid in the first place, only then load store when we already know that dbType is valid

File: tools/cli/adminDBScanCommand.go [hunk 3]
```diff
@@ -350,3 +369,32 @@ func getSQLParser(encodingType common.EncodingType, decodingTypes ...common.Enco
...
+	decodingTypesStr := c.StringSlice(FlagDecodingTypes)
+	var decodingTypes []common.EncodingType
+	for _, dt := range decodingTypesStr {
+		decodingTypes = append(decodingTypes, common.EncodingType(dt))
+	}
+	shardStore, err := sql.NewShardPersistence(sqlDB, currentClusterName, logger, getSQLParser(common.EncodingType(encodingType), decodingTypes...))
+	if err != nil {
+		ErrorAndExit("Failed to get shard store from sql config", err)
+	}
+	return shardStore
```
Comment (mantas-sidlauskas) [inline tools/cli/adminDBScanCommand.go:399]: Maybe return err also? Exit paths could be stored in a single place

File: tools/cli/adminDBScanCommand.go [hunk 4]
```diff
@@ -316,6 +316,25 @@ func initializeHistoryManager(c *cli.Context) persistence.HistoryManager {
...
+	default:
+		if sql.PluginRegistered(dbType) {
+			shardStore = initializeSQLShardStore(c, "current-cluster", logger)
+		} else {
+			supportedDBs := append(sql.GetRegisteredPluginNames(), "cassandra")
+			ErrorAndExit(fmt.Sprintf("The DB type is not supported. Options are: %s.", supportedDBs), nil)
+		}
+	}
+	shardManager := persistence.NewShardManager(shardStore)
+	return shardManager
```
Comment (mantas-sidlauskas) [inline tools/cli/adminDBScanCommand.go:335]:
```suggestion
	return persistence.NewShardManager(shardStore)
```

**Review by mantas-sidlauskas — COMMENTED** [review]
Some comments for your consideration

File: tools/cli/adminDBScanCommand.go (hunk 3 above)
Comment (Shaddoll) [reply tools/cli/adminDBScanCommand.go:394]: It's to keep the function signature the same as others.

**Review by yycptt — APPROVED** [review]
Please do not commit the cli binary (at cmd/toosl/cli/cli)

**Pull request merged**